src/main/java/com/mongodb/hibernate/dialect/MongoDialect.java

```diff
@@ -39,11 +42,40 @@ public MongoDialect(DialectResolutionInfo info) {
         super(info);
     }
 
+    /**
+     * This constructor is called only if Hibernate ORM falls back to it due to a failure of
+     * {@link MongoDialect#MongoDialect(DialectResolutionInfo)}.
+     *
+     * @deprecated Exists only to avoid the confusing {@link NoSuchMethodException} thrown by Hibernate ORM when
```

Comment:
I am happy for this change for yeah, it confuses everybody when mongodb is not up (like first-time code reviewers lik Maxim).